Export as pip package

kanevk opened the pull request:
The PR does the following: 
 - [x] move the config to ~/gkeep.config.json
 - [x] add package structure and setup.py
 - [x] add a script that generates the config
 - [x] Add the instructions for usage as a package to the README

kanevk committed:
Convert to a package

 - move the config to HOME/gkeep.config.json
 - add package structure and setup.py
 - add script that generates the config

kanevk committed:
Attach the new release

kanevk commented on README.md:
```diff
@@ -1,7 +1,7 @@
 
 - Helps you backup your local file notes
-- Helps you easier access your Google Keep notes through your favorite text editor
+- Helps you access your _Google Keep_ notes through the local file system and therefore your favorite text editor
```
```suggestion
- Helps you access your _Google Keep_ notes through the local file system
```

kanevk commented on gkeep_sync/sync_note.py:
```diff
@@ -0,0 +1,163 @@
+FILE_EXTENSION = 'md'
+
+
+class GkeepSyncAPI:
```
I have to make a bunch of methods private ones

scdekov commented on gkeep_sync/sync_note.py:
```diff
@@ -0,0 +1,163 @@
+    def create_note(self, title, text):
+        print(f"Creating a note {title}")
+        newNote = self.keep.createNote(title, text)
+        label = self.keep.findLabel(SYNC_LABEL)
```
should you get this label only once(probably in __init__)?

kanevk replied:
fair enough

scdekov commented on gkeep_sync/sync_note.py:
```diff
@@ -0,0 +1,163 @@
+
+        if len(gnotes) > 1:
+            raise Exception(f'Too many notes match the title: {title}')
+        return (gnotes[0] if len(gnotes) == 1 else None)
```
No need for parentheses here. You can also change `len(gnotes) == 1` to just `gnotes` since you already know that length is eight 0 or 1

kanevk replied:
I prefer the version with the parens.

scdekov commented on gkeep_sync/sync_note.py:
```diff
@@ -0,0 +1,163 @@
+
+
+    @benchmark
+    def run(self, note_path):
```
That method should probably be named sync_note/file instead of run

kanevk committed:
Freshen the README

kanevk committed:
Find the auto sync lable only once

kanevk committed:
Improve style

kanevk committed:
Improve style and interfaces

 - add static method from login + api initiation
 - add private methods

kanevk committed:
Rename sync_note to sync_api

kanevk committed:
Save the state file into user's home

When the code is executed as a package we cannot rely to create the
state file into the executable's directory.

We'll preserve it in `~/.gkeep-state.json`

scdekov commented on gkeep_sync/sync_server.py:
```diff
@@ -3,47 +3,45 @@
-    sync_note.login(keep)
 
-    sync_note.upload_new_notes(keep)
+    sync_api = SyncAPI.login()
```
it's a little confusing to me that the login method is returning an instance

kanevk replied:
Yes, I share that, but this have several benefits as well:
 - The public interface is minimal
 - Since the SyncApi is dysfunctional without `login` it makes sense to don't have access to the instance without logging in
